src/DataBus/AzureDataBusPersistence.cs

```diff
@@ -14,11 +20,63 @@ protected override void Setup(FeatureConfigurationContext context)
         {
             var dataBusSettings = context.Settings.GetOrDefault<DataBusSettings>() ?? new DataBusSettings();
 
-            var cloudBlobClient = CloudStorageAccount.Parse(dataBusSettings.ConnectionString).CreateCloudBlobClient();
+            ThrowIfConnectionStringAndTokenProviderSpecified(dataBusSettings);
 
-            var dataBus = new BlobStorageDataBus(cloudBlobClient.GetContainerReference(dataBusSettings.Container), dataBusSettings, new AsyncTimer());
+            var container = CreateCloudBlobContainer(dataBusSettings);
+
+            var dataBus = new BlobStorageDataBus(container, dataBusSettings, new AsyncTimer());
 
             context.Container.ConfigureComponent(b => dataBus, DependencyLifecycle.SingleInstance);
         }
+
+        static void ThrowIfConnectionStringAndTokenProviderSpecified(DataBusSettings dataBusSettings)
+        {
+            if (dataBusSettings.RenewalTimeBeforeTokenExpires != null && dataBusSettings.UserProvidedConnectionString)
+            {
+                throw new Exception("More than one method to connect to the storage account was supplied (using connection string and token provider). Use one method only.");
```

Comment:
Does it make sense to have a more specific exception here?

Reply:
I see that the error message has not been updated 😂
Will udpate.

Reply:
I mean a specific type instead of throwing the `Exception` type.

Reply:
Used to do that in the past, but since you can't recover from the exception, there was not much benefit of a specific exception.